Modules/FV0/src/TriggersSwVsTcmCheck.cxx

```diff
@@ -57,14 +78,19 @@ Quality TriggersSwVsTcmCheck::check(std::map<std::string, std::shared_ptr<Monito
       auto* histogram = dynamic_cast<TH2F*>(mo->getObject());
       result = Quality::Good;
       int numberOfBinsX = histogram->GetNbinsX();
-      for (int i = 1; i <= numberOfBinsX; i++) {
-        if ((bool)(histogram->GetBinContent(i, kBinSwOnly)) == (bool)(histogram->GetBinContent(i, kBinTcmOnly))) {
+      for (int binId = 1; binId <= numberOfBinsX; binId++) {
+        if ((bool)(histogram->GetBinContent(binId, kBinSwOnly)) != (bool)(histogram->GetBinContent(binId, kBinTcmOnly))) {
```

Comment:
I think what should raise an error should be:

`histogram->GetBinContent(binId, kBinSwOnly) || histogram->GetBinContent(binId, kBinTcmOnly)`

`kBinSwOnly` means that an event "fired a software trigger" but NOT the TCM trigger, and vice versa for `kBinTcmOnly`. If an event fires only one of these, there is a problem, they should always match (i.e. both fired, or neither fired, is fine). Then we might discuss what the threshold for raising the error should be, i.e. a small amount of events with problems might be fine (?), but for now, you can make a hard cut and don't allow any problems.

You're current check wouldn't catch the situation when swOnly > 0 && tcmOnly > 0

Reply:
@andreasmolander we should always rise the error when swOnly > 0 || tcmOnly > 0.

Reply:
Done